### ISSUE TITLE
Firmware Problem

### ISSUE THREAD

**Issue by don-teka** [lukasz1992/asuswrt-xiaomi#39]
Hey what happened to the firmware? the last 2 versions do not boot the MIR3G, it seems that it got worse instead of getting better. I tested the last 2 compilations through the breed does not work, the router does not start, and the method you described a while ago in the firmware 3.0.0.4_382_52107 also does not work the router simply does not start, the same happens if I update directly through the interface web.

**Comment by lukasz1992** [comment]
Hi @don-teka, I am sorry that you have problems. I decided to drop support for dual image, as I really want to minimise own implementation. I am also not sure what env variables you have provided in Breed.

Please:
- Flash 3.0.0.4_382_52107 this way https://github.com/lukasz1992/asuswrt-xiaomi/issues/34#issuecomment-640719152
- Then upgrade to 3.0.0.4_382_52111 with User Interface (Upgrade Firmware page)

**Comment by don-teka** [comment]
So, yesterday I tested several ways to install the latest build, including the method you suggested starting from version 52107 after upgrading to version 52111 and the router just doesn't start with that latest build. I even used the method you mentioned directly with this latest version 52111 and I was unsuccessful, I waited about 3 minutes but the front LED didn't even light up. Through the interface in the firmware update part the same problem, being in version 52107 when updating the 52111 firmware the router restarts, flashes the blue LED only once, and turns off and does not start again. I think there is something wrong with the boot in this last build.

**Comment by lukasz1992** [comment]
Something must be wrong with your Breed. Could you reflash it (erasing its config)?
Do you have bad sectors on NAND? Do you have log from serial console?

**Comment by H3MKL** [comment]
Hello, I am using r2100, I tried to use breed and telnet flashing,  it failed to start, but the r3g firmware is normal, please tell me the correct refresh method

**Comment by adamnickster** [comment]
I have the same problem as @don-teka. 
In the past, I always got boot older images only without boot flash 0x600000 in breed settings(flashed firmware to both partitions). 
New release 52111 does not boot at all. 
Debug info: 
If I set older functional firmware to boot from 0x600000 it does not work. 4pda.ru site tells to boot asuswrt from the first partition that works for me.

Wifi speed is slow on 52107( iPhone 6s and iPhone 7 upload is only ~9Mbps before ~100mbps   witch is Full link speed.

**Comment by lukasz1992** [comment]
@adamnickster tell exactly which versions works for you (52111, 52110, 52107).

My current recommendations are the following:
- partition at 0x200000 on NAND should be erased and empty
- nvram flag_try_sys1_failed must be set to 1 

I do not have account on 4pda.ru, and can not be responsible what they tell about asuswrt.

@adamnickster please also open a separate ticket for WiFi problems

**Comment by lukasz1992** [comment]
EDIT: I confirmed 52111 is faulty, I will update 52111 images when I fix them. Sorry for that

**Comment by adamnickster** [comment]
@lukasz1992 thank you.

Can you write short how-to to erase all settings/nvram/partitions/etc...? (maybe mtd command or breed flash command?)

OT: (After factory reset(GUI) with checked erase all data, still persist some users in /etc/passwd and samba not working because /etc/samba/smbpasswd is empty after each restart of samba. I want the clean state of firmware and router.)

**Comment by lukasz1992** [comment]
On previous releases, kernel relied on bootloader to provide correct rootfs partition. However this implementation did not work on OEM bootloader and kernel crashed very early (before printing anything).

So I used Mediatek version of cmdline, which did not work on NAND devices (wrong partition offset, because Factory2 partition was not taken into account). I fixed it in 2f03060, and images should work now correctly. Please wait about 30 minutes for fixed images.

**Comment by lukasz1992** [comment]
@adamnickster I did not have such problem, actually I have never tested Samba. Maybe Asus implementation is broken, don't know about it.
You run do these commands (without mounted USB disk):
```
mtd-erase -d nvram
mtd-erase -d jffs2
reboot -f
```

**Comment by adamnickster** [comment]
> On previous releases, kernel relied on bootloader to provide correct rootfs partition. However this implementation did not work on OEM bootloader and kernel crashed very early (before printing anything).
> 
> So I used Mediatek version of cmdline, which did not work on NAND devices (wrong partition offset, because Factory2 partition was not taken into account). I fixed it in [2f03060](https://github.com/lukasz1992/asuswrt-xiaomi/commit/2f03060dfe081cc32622d85d13cda38d7d805820), and images should work now correctly. Please wait about 30 minutes for fixed images.

thanks, tested and worked

**Comment by don-teka** [comment]
@adamnickster  sorry but did you test how? I had already given up, but I saw that there was a correction, but I didn't see anything new in the release.

**Comment by adamnickster** [comment]
@don-teka downloaded image from latest release and flashed it through web interface.(Lukasz just updated download link on that release)
